packages/integration-objects/srv/utils/AuditLog.js

```diff
@@ -0,0 +1,100 @@
+/* Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. */
+'use strict';
+const xsenv = require('@sap/xsenv');
+const auditLogging = require('@sap/audit-logging');
+const { logger } = require('../logging/Logger');
+
+const credentials = xsenv.getServices({ auditlog: { tag: 'auditlog' } }).auditlog;
+const xsuaa = xsenv.getServices({ xsuaa: { tag: 'xsuaa' } }).xsuaa;
+
+class AuditLog {
+  constructor() {
+    this.tenantId = xsuaa ? xsuaa.tenantid : null;
+    console.log(this);
+
+    auditLogging.v2(credentials, (err, auditLog) => {
+      if (err) {
+        logger.error('Failed to instantiate audit log', err);
+      } else {
+        this.auditLog = auditLog;
+      }
+    });
+  }
+
+  getAuditLog() {
+    return this.auditLog;
+  }
+
+  dataModificationMessage() {
+    if (this.auditLog) {
+      return this.auditLog
+        .update(...arguments)
+        .dataSubject({ type: 'system', id: { key: this.getSelfTenantId() } })
+        .by('system')
+        .tenant(this.getSelfTenantId());
+    }
+    return null;
+  }
+
+  configurationChange() {
+    if (this.auditLog) {
+      return this.auditLog
+        .configurationChange(...arguments)
```

Comment:
where does the `arguments` variable come from?

Reply:
Since I wrote the AuditLog wrapper, I made it generic to accept whatever arguments are passed. However, in our case, as of now, we are just passing one argument. It's being passed by the writeAuditLog method in the AuditLogWriter file

Reply:
ok then please add the argument in the method signature in line 39. The same for all other methods as well. If you expect and argument tell it to the user. 

Also please add a issue in the backlog to transform the `utils,validations,logging` to TypeScript such that we have a more strict control.

Reply:
updated in the issues.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments

This is the arguments variable that I am using. The wrapper is made generic to accept any number of change arguments that have to be audited. For eg., if I am auditing 5 different change attributes, I'd not need to change the wrapper.

Reply:
Hey @vidhyakpai,

as discussed, did you try to do something like this:
```ts
configurationChange(...args) {
  return this.auditLog.configurationChange(...args)
}
```

That way the signature of you function reveals, that you are using the parameters.
Even the documentation that you shared, states: "Note: If you're writing ES6 compatible code, then rest parameters should be preferred."

Reply:
@marikaner , this is done and PR is updated.